src/types/EmbeddedPaymentElement.tsx

```diff
@@ -0,0 +1,427 @@
+import {
+  requireNativeComponent,
+  NativeModules,
+  NativeEventEmitter,
+  EmitterSubscription,
+  LayoutAnimation,
+} from 'react-native';
+import type {
+  BillingDetails,
+  AddressDetails,
+  UserInterfaceStyle,
+  CardBrand,
+} from './Common';
+import type { PaymentMethod } from '.';
+import * as PaymentSheetTypes from './PaymentSheet';
+import type { ImageSourcePropType, ViewProps } from 'react-native';
+import NativeStripeSdk from '../NativeStripeSdk';
+import {
+  ReactElement,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import React from 'react';
+
+// Native bridge imports
+const { StripeSdk } = NativeModules;
+const NativeStripeEmbedded = StripeSdk;
+const eventEmitter = new NativeEventEmitter(NativeModules.StripeSdk);
+
+// -----------------------------------------------------------------------------
+// Types
+// -----------------------------------------------------------------------------
+
+/**
+ * The final result of a confirm call.
+ * Typically: payment succeeded (“completed”), canceled, or failed with error.
+ */
+export type EmbeddedPaymentElementResult =
+  | { status: 'completed' }
+  | { status: 'canceled' }
+  | { status: 'failed'; error: Error };
+
+/**
+ * The result of an update call.
+ */
+export type EmbeddedPaymentElementUpdateResult =
+  | { status: 'succeeded' }
+  | { status: 'canceled' }
```

Comment:
Android doesn't have canceled as this would only happen if the merchant cancelled via a kotlin coroutine cancelation.

Reply:
I suppose it's reasonable to never pass back canceled when used on Android then.